src/useAsync.ts

```diff
@@ -12,26 +12,25 @@ const useAsync = <T>(fn: () => Promise<T>, args?) => {
   });
   const memoized = useCallback(fn, args);
 
-  useEffect(() => {
+  useEffect(async () => {
```

Comment:
I don't think `useEffect` can be an `async` function, because it has to return the unsubscription function below, or can it be async?